archon-ui-main/src/services/knowledgeBaseErrorHandler.ts

```
@@ -0,0 +1,220 @@
+}
+
+/**
+ * Get suggested action for the user based on error type
+ */
+export function getErrorAction(error: EnhancedError): string | null {
+  if (error.isOpenAIError && error.errorDetails) {
+    switch (error.errorDetails.error_type) {
+      case 'quota_exhausted':
+        return 'Check your OpenAI billing dashboard and add credits';
+      case 'rate_limit':
+        const retryAfter = error.errorDetails.retry_after || 30;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)